modules/core/src/v2/coins/hbar.ts

```diff
@@ -146,7 +155,18 @@ export class Hbar extends BaseCoin {
   }
 
   isWalletAddress(params: VerifyAddressOptions): boolean {
-    throw new MethodNotImplementedError();
+    const { address, baseAddress } = params;
+    if (!this.isValidAddress(address)) {
+      throw new InvalidAddressError(`invalid address: ${address}`);
+    }
+    const addressDetails = this.getAddressDetails(address);
+    const rootAddressDetails = this.getAddressDetails(baseAddress);
+    if (addressDetails.address !== rootAddressDetails.address) {
+      throw new UnexpectedAddressError(
+        `address validation failure: ${addressDetails.address} vs ${rootAddressDetails.address}`
+      );
+    }
+    return true;
```

Comment:
please deduplicate this code (maybe `hasSameBaseAddress()` in a new file `coins/address/index.ts`)